### ISSUE TITLE
Rolling restart logic is broken

### ISSUE THREAD

**Issue by saketmht** [Opster/opensearch-k8s-operator#604]
After configuring my cluster CR (by adding a wrong plugin name), the operator updates the sts with the wrong plugin name and restarts one the pods (rolling restart) but the pod is stuck in CLB. When I correct the the plugin name in the CR, the sts is updated with the right name but the pods are not restarted. This is likely due to multiple checks that we have before performing a rolling restart of the pods.

Before doing a rolling restart:

- We check if all the data nodes pods are ready or not. This would not be the case on the second try(when the plugin name is corrected) since one of the pod is in CLB

- Even after this check, we again check if the sts is healthy i.e. replicas == readyReplicas. Again this would not be the case as one of the pods is stuck in CLB.

While this can easily be fixed by deleting the pod manually, the operator should be handling this case rather the user having to do anything.

Cluster CR -
```
apiVersion: opensearch.opster.io/v1
kind: OpenSearchCluster
metadata:
  name: vz-os-cluster
  namespace: default
spec:
  security:
    tls:
      transport:
        generate: true
      http:
        generate: true
  general:
    httpPort: 9200
    serviceName: vz-os-cluster
    version: 2.3.0
    drainDataNodes: false
  dashboards:
    tls:
      enable: true
      generate: true
    version: 2.3.0
    enable: true
    replicas: 1
    resources:
      requests:
         memory: "512Mi"
         cpu: "200m"
      limits:
         memory: "512Mi"
         cpu: "200m"
  nodePools:
    - component: masters
      replicas: 3
      resources:
         requests:
            memory: "1Gi"
      roles:
        - "data"
        - "cluster_manager"
      persistence:
        emptyDir: { }
```
After all the pods are ready, add a wrong plugin to the pluginsList and the pod goes to CLB. Correct the plugin name but the pods are restarted anymore.

**Comment by saketmht** [comment]
Misconfiguring the plugin name is just one example. If for any reason, one of the pods is stuck on a rolling restart, then it seems any further updates to the CR won't have any effect as it will keep on waiting for sts to be healthy before attempting another restart.

**Comment by swoehrl-mw** [comment]
Should be fixed with PR #614
